### PR TITLE
New action for checking the format of pull requests

### LINKED ISSUE #3: Check pull request format
### Aim

This task aims at implementing a GitHub Actions script for checking that a pull request (PR) follows some rules concerning its title contents. This action is applied more specifically to squash and merge PRs and repositories applying the squash and merge strategy to keep their git commits history clean and lean, without using [Conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).

The format of a PR's title must be of the form:

 `Modify the data model (#3)`

Where the index in parenthesis (i.e. `(#3)`) refers to the number of the PR containing the (inner) commits when later the PR is merged to the main branch.

It also aims to verify the comits' message inside the PR to be reliable for the `semantic-release`work.
The message must be of the form:

`feat(github): change the automatic labeling`

### Assumptions

- None

### Acceptance criteria

- [x] The new GitHub Actions script is named `check-pull-request-format-action`
- [x] Only the title of the squash and merge PR commit message must be analyzed
- [x] The new GitHub Actions script must be implemented in Typescript 

### Risks

- None

### Dependencies

- None

### References

- [Creating a Javascript action](https://docs.github.com/en/actions/creating-actions/creating-a-javascript-action)
- [Typescript GitHub Actions boilerplate](https://github.com/actions/typescript-action)

### PR DESCRIPTION
**Pull request by binadamu-isiyoonekana**
## Description

This pull request (PR) implements a GitHub Actions script for checking that a pull request is well-formed before being merged to the main branch of a repository.

It particularly targets repositories using the squash and merge commit strategy so that to keep their commit history clean and readable.

It also targets repositories using the [`semantic-release`](https://github.com/semantic-release/semantic-release) tool for automatically generating releases based on the [conventional commits](https://github.com/conventional-commits-rs) standard. Here, only non-conventional (squash and merge) commits are expected in the main branch, each with a reference to the pull request it was squashed from, allowing to refer to the inner-commits, though the latter were [squashed](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-commits) during the merging of the pull request to the main branch. The format of such squash and merge commits is as follows:

```text
Modify the data model (#13)
```

where the block `(#13)`, and particularly the number `13`, is the index of the pull request containing the commits that were squashed and merged.

## Related issues

This PR closes issue #3 

## How has this been tested?

Unit tests are executed using `npm test` command inside the subfolder of the action.

**Test configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] Action is successfully tested with [Act](https://github.com/nektos/act)
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: .github/workflows/test-check-pull-request-action.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    name: Test action
+    runs-on: ubuntu-latest
+
+    defaults:
+      run:
+        working-directory: ./check-pull-request-action
+
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v2
```
Comment (GuiBL4) [inline .github/workflows/test-check-pull-request-action.yml]: Is-it better to use checkout@v3 here ?

File: check-pull-request-action/README.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+```bash
+act -s GITHUB_TOKEN=[Your_Personal_Github_Token] pull_request_target
+```
+
+## References
+
+- [Typescript Action boilerplate](https://github.com/actions/typescript-action)
+- Hassan SANI. June 6th 2022. [Creating your own GitHub action with TypeScript](https://www.thisdot.co/blog/creating-your-own-github-action-with-typescript)
+- https://github.com/ktrz/gh-action-js-marathon/blob/feat/setup-action/.github/workflows/test.yml
```
Comment (GuiBL4) [inline check-pull-request-action/README.md:90]: hyperlink is not needed here ?

File: check-pull-request-action/src/parse.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+/**
+ * Parses the subject line of a pull request
+ *
+ * This function analyses messagesquash and merge commit combines several commits from a pull request's branch to the main branch, so
```
Comment (GuiBL4) [inline check-pull-request-action/src/parse.ts]: It seems there is a problem with this sentence.

File: check-pull-request-action/src/parse.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+ * that to keep the git history lean and clean.
+ * The expected format of a conventional squash and merge message is as follows:
+ *
+ *   'Modify the data model (#34)'
+ *
+ * Ehere '34' is the number of the pull request the input squash and commit message refers
```
Comment (GuiBL4) [inline check-pull-request-action/src/parse.ts]: Little typo here : 'Ehere' ?

File: check-pull-request-action/README.md (hunk 2 above)
Comment (binadamu-isiyoonekana) [reply check-pull-request-action/README.md:90]: Actually they are mentioned

**Review by GuiBL4 — APPROVED** [review]

**Pull request merged**